test/spec/windshaft/client.spec.js

```diff
@@ -273,4 +273,30 @@ describe('windshaft/client', function () {
       });
     });
   });
+
+  describe('instantiateMap', function () {
+    var client;
+    beforeEach(function () {
+      client = new WindshaftClient({
+        urlTemplate: 'https://{user}.example.com:443',
+        userName: 'rambo'
+      });
+    });
+
+    it('should make a request when the request service allows it', function () {
+      spyOn(client, '_instantiateMap').and.callFake(function () {});
+      spyOn(client._requestTracker, 'canRequestBePerformed').and.returnValue(true);
```

Comment:
If we really need to spy on private methods, I'd at least abstract this (`client._requestTracker.canRequestBePerformed`) in a method (as it was before) so that we don't have to know about the internals of the `RequestTracker` here...